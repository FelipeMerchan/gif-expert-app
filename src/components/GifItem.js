import React from 'react';

export const GifItem = ({ title, url }) => {
  return (
    <div>
      <img
        src={url}
        alt={title}
        title={title}
      />
      <p>{title}</p>
    </div>
  )
}
