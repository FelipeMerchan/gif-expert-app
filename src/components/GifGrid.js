import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';

import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {

  const { data: images, loading } = useFetchGifs(category);

  return (
    <section aria-label="Gif list">
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && <p className="animate__animated animate__flash">Cargando...</p>}

      <div className="card-grid">
        {
          images.map(img => (
            <GifItem
              key={img.id}
              {...img}
            />
          ))
        }
      </div>
    </section>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
