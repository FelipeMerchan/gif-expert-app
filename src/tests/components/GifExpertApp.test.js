import { shallow } from 'enzyme';
import { GifExpertApp } from '../../components/GifExpertApp';

describe('Pruebas en el componente <GifExpertApp />', () => {
  test('Debe mostrar el componente correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar una lista de categorÃ­as', () => {
    const categories = ['One Punch Man', 'Demon Slayer'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});