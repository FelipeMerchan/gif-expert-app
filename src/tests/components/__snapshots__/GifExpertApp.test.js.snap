// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch Man"
      key="One Punch Man"
    />
    <GifGrid
      category="Demon Slayer"
      key="Demon Slayer"
    />
  </ol>
</Fragment>
`;
